### ISSUE TITLE
Freqtrade trailing stop loss exit active while set to false

### ISSUE THREAD

**Issue by adsjr1** [freqtrade/freqtrade#7083]
* Operating system: Windows
  * Python Version: 3.10.5 
  * CCXT version: 1.89.96 
  * Freqtrade Version: develop
  
I checked issue #7055 and it is not the same thing.
I noticed that backtesting was giving me many trailing stop loss exits already. I am doing a dry run of my strategy  and I have just received a trailing stop loss exit

### Steps to reproduce:

  1. Started the dry run
  2. Freqtrade entered a trade
  3. Trailing stop loss exit came after reaching below 1% (???)
  
### Observed Results:

  * What happened?
  Got this message on my telegram bot:
Profit: -1.01% (loss: -0.91270320 USDT / -0.914 USD)
Enter Tag: None
Exit Reason: trailing_stop_loss
Duration: 0:34:00 (34.0 min)
Direction: Long
Amount: 1048.00000000
Open Rate: 0.08580000
Close Rate: 0.08510000

 * What did you expect to happen?
 No message as the trade should only exit on a custom stop loss

### Relevant code exceptions or logs

  ```
    INTERFACE_VERSION = 3

    # Optimal timeframe for the strategy.
    timeframe = '5m'

    # Can this strategy go short?
    can_short: bool = False

    # Minimal ROI designed for the strategy.
    # This attribute will be overridden if the config file contains "minimal_roi".
    minimal_roi = {
       "0": 100
    }

    custom_info = {
        'risk_reward_ratio': 2.0,
        'set_to_break_even_at_profit': 1,
    }
    
    # Optimal stoploss designed for the strategy.
    # This attribute will be overridden if the config file contains "stoploss".
    use_custom_stoploss = True
    stoploss = -0.99

    # Trailing stoploss
    trailing_stop = False
    # trailing_only_offset_is_reached = False
    # trailing_stop_positive = 0.01
    # trailing_stop_positive_offset = 0.0  # Disabled / not configured

    # Run "populate_indicators()" only for new candle.
    process_only_new_candles = False

    # These values can be overridden in the config.
    use_exit_signal = True
    exit_profit_only = False
    ignore_roi_if_entry_signal = False

    # Number of candles the strategy requires before producing valid signals
    startup_candle_count: int = 200

def custom_stoploss(self, pair: str, trade: 'Trade', current_time: datetime,
                        current_rate: float, current_profit: float, **kwargs) -> float:

        """
            custom_stoploss using a risk/reward ratio
        """
        result = break_even_sl = takeprofit_sl = -1
        custom_info_pair = self.custom_info.get(pair)
        if custom_info_pair is not None:
            # using current_time/open_date directly via custom_info_pair[trade.open_daten]
            # would only work in backtesting/hyperopt.
            # in live/dry-run, we have to search for nearest row before it
            open_date_mask = custom_info_pair.index.unique().get_loc(trade.open_date_utc, method='ffill')
            open_df = custom_info_pair.iloc[open_date_mask]

            # trade might be open too long for us to find opening candle
            if(len(open_df) != 1):
                return -1 # won't update current stoploss

            initial_sl_abs = open_df['stoploss_rate']
            
            # calculate initial stoploss at open_date
            initial_sl = initial_sl_abs/current_rate-1

            # calculate take profit treshold
            # by using the initial risk and multiplying it
            risk_distance = trade.open_rate-initial_sl_abs
            reward_distance = risk_distance*self.custom_info['risk_reward_ratio']
            # take_profit tries to lock in profit once price gets over
            # risk/reward ratio treshold
            take_profit_price_abs = trade.open_rate+reward_distance
            # take_profit gets triggerd at this profit
            take_profit_pct = take_profit_price_abs/trade.open_rate-1

            # break_even tries to set sl at open_rate+fees (0 loss)
            break_even_profit_distance = risk_distance*self.custom_info['set_to_break_even_at_profit']
            # break_even gets triggerd at this profit
            break_even_profit_pct = (break_even_profit_distance+current_rate)/current_rate-1

            result = initial_sl
            
            if(current_profit >= break_even_profit_pct):
                break_even_sl = (trade.open_rate*(1+trade.fee_open+trade.fee_close) / current_rate)-1
                result = break_even_sl

            if(current_profit >= take_profit_pct):
                takeprofit_sl = take_profit_price_abs/current_rate-1
                result = takeprofit_sl

       return result


  ```

**Comment by xmatthias** [comment]
Custom stoploss acts as dynamic trailing stoploss. 

The difference between the two is who sets the stoploss (static / your function) - but the remaining logic is identical, therefore the exit reason will also be identical.

We also recommend to only use one of the two types, so differentiating between the two is not really relevant.

**Comment by adsjr1** [comment]
I found out what is going on. It looks like my custom stop loss is making a trailing stop loss, although I wanted a fixed value. I will keep observing it to confirm. Thanks!

**Comment by adsjr1** [comment]
Looking at the code again I don't think it is a trailing stop loss. In the indicators section this is what I have regarding stoploss:
```
# Stoploss data

        dataframe['stoploss_rate'] = dataframe[['st_long_trend','st_short_trend']].max(axis=1)
        self.custom_info[metadata['pair']] = dataframe[['date', 'stoploss_rate']].copy().set_index('date')
```
The 'stoploss_rate' column will be updated after each candle, but in the custom stop loss function, see that the stop loss value is the one when the trade opened (initial_sl_abs = open_df['stoploss_rate']):

```
if custom_info_pair is not None:
            # using current_time/open_date directly via custom_info_pair[trade.open_daten]
            # would only work in backtesting/hyperopt.
            # in live/dry-run, we have to search for nearest row before it
            open_date_mask = custom_info_pair.index.unique().get_loc(trade.open_date_utc, method='ffill')
            open_df = custom_info_pair.iloc[open_date_mask]

            # trade might be open too long for us to find opening candle
            if(len(open_df) != 1):
                return -1 # won't update current stoploss

            initial_sl_abs = open_df['stoploss_rate']
```
Does it make sense what I am saying? I just copied the risk-reward function from the advanced strategies folder.

**Comment by xmatthias** [comment]
Don't neglect the possibility that you have trailing_stop activated from the configuration. startup-logs will show this. It's unlikely but a possibility, depending on the configuration used.

I'd not trust your break_even_profit_pct section - it seems wrong at first glance. 
With the provided info, it's pretty difficult to give you more advice without fully debugging / running through your code - which  i don't plan on doing.

in the end, it'll be your strategy that returned this value - however this could be a -0.02 (-2%) when the trade was still at +1% profit - which might be expected (but it really isn't).

The return value is simply the relative value from the current rate. 
for the calculations themselves i'd strongly recommend to use the provided [helpers](https://www.freqtrade.io/en/stable/strategy-callbacks/#common-helpers-for-stoploss-calculations) - which will avoid some odd errors (especially once you deal with shorts as well - which will need to use a slightly different calculation).

**Comment by xmatthias** [comment]
I'll assume this as solved.

Feel free to reopen / comment below if that's not the case.
